### PR TITLE
[Single Variation] Add missing tracks events

### PR DESCRIPTION
**Pull request by octaedro**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Add a couple of missing Tracks events:

- `wcadmin_product_toggle_click`: with the props `block_id: "product-variation-track-stock"` and `block_id: "product-variation-virtual"`

Rename prop `description` to `note` for single variations.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a WooCommerce store with this branch and enable the product editor using **WooCommerce > Settings > Advanced > Features**
2. Go to **Products > Add New**
3. Go to `/wp-admin/admin.php?page=wc-admin&path=/add-product&tab=variations` and add some attributes
4. Wait until variations get generated
5. Go to Tools > WCA Test Helper > Tools and run the `Enable wc-admin Tracking`
6. Open the browser toolkit and go to the `Console` tab
7. Press the `Edit` button in one of the variations
8. Go to the `Inventory` tab and press the `Track stock quantity for this product` toggle
9. The event: wcadmin_product_toggle_click should be tracked with the prop `block_id: "product-variation-track-stock"` (it should appear in the console opened in step 6).
10. Go to the `Shipping` tab and press the `This variation requires shipping or pickup` toggle
11. The event: wcadmin_product_toggle_click should be tracked with the prop `block_id: "product-variation-virtual"`.
12. Go to the `General` tab and add a text under `Note`
13. Press `Update` and verify that the event `wcadmin_product_update` is recorded with the prop `note: "yes"`
14. Press `< Main product`, add a description under the `General` tab and press `Update`
15. Verify that the event `wcadmin_product_update` is recorded with the prop `description: "yes"` 

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/components/header/publish-button/publish-button.tsx [hunk 1]
```diff
@@ -36,7 +36,19 @@ export function PublishButton( {
...
+				> &
+					Partial< Product >;
+				if ( savedProduct.parent_id > 0 ) {
+					const { description, ...rest } = savedProduct;
+					propsToRecord = {
+						...rest,
+						note: description,
+					};
+				}
+				recordProductEvent( 'product_update', propsToRecord );
```
Comment (mdperez86) [inline packages/js/product-editor/src/components/header/publish-button/publish-button.tsx]: This logic related to the data we are going to track must be implemented inside the `recordProductEvent`

**Review by mdperez86 — COMMENTED** [review]
Nice job @octaedro I left one comment

File: packages/js/product-editor/src/components/header/publish-button/publish-button.tsx (hunk 1 above)
Comment (octaedro) [reply packages/js/product-editor/src/components/header/publish-button/publish-button.tsx]: Using a variable for `recordEvent` props is [a pattern that we are already using](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/homescreen/activity-panel/index.js#L93-L102), but I'm okay with changing it.

Comment (octaedro) [reply packages/js/product-editor/src/components/header/publish-button/publish-button.tsx]: Scratch my last comment. I got what you meant. I fixed it in the commit [5f4b8f9](https://github.com/woocommerce/woocommerce/pull/40996/commits/5f4b8f9cd516c6ef9f8f1c0091d0d118ea330d3a)

**Review by AnnaMag — APPROVED** [review]
LGTM

**Review by mdperez86 — APPROVED** [review]
Nice job @octaedro LGTM !!!

**Pull request merged**